### PR TITLE
Anim Tester X - Quality of Life improvements

### PR DESCRIPTION
**Pull request by LuminarLight**
I started using the Anim Tester X in Jak II, and noticed it could use some QoL improvements. So I implemented the following things:

"Refreshing" the art groups is now possible through the debug menu. The way I implemented it may do more cleanup than required, but it should be safe at least.

Now you can cycle through animations using the dpad, without having to go back to the debug menu. I only implemented it for dpad left and dpad right, each moving the selection by 1. I didn't see valid use case for adding bigger jumps (like in the debug menu), because outside the menu we can't see the anim list anyway.

Known issues:
Moving between levels without stopping Anim Tester X can lead to issues, but I don't think this was caused by me.
Cycling through animations too fast can crash the game. Again I don't really think this is my fault, but still it is worth mentioning.

Feedback is very welcome. I tried to keep the code format consistent with the project. Also it is possible that what I did could be done better by more-experienced people....

And in case this gets accepted:
I should implement these same changes to TPL before the PR is merged, because anim tester x is pretty much the same there. But first I want to see if the changes I did in Jak II are even okay.

### CODE REVIEW

**Comment by ManDude** [comment]
The whole thing needs to be rewritten because it hasn't worked properly since the new merc renderer.

**Comment by LuminarLight** [comment]
What are the issues with it? Maybe I didn't use it properly/enough, but I didn't see any serious issues. The only thing I noticed that is maybe an issue is that art groups with "components" (not sure what they are) have all components appear at the same time (for example all clothes/beards/hairs of citizen-fat), which is maybe not the intended behaviour.

But if I understand correctly, you want to have the whole Anim Tester X™ rewritten? Would these changes still be acceptable for now? Because I imagine the rewrite won't happen very soon, and until then these changes can improve the experience.

**Review by ManDude — CHANGES_REQUESTED** [review]

File: goal_src/jak2/pc/debug/default-menu-pc.gc [hunk 1]
```diff
@@ -495,6 +495,14 @@
     )
   #f)
 
+(defun dm-anim-tester-x-refresh ()
+  (atx-list-init! (-> *ATX-settings* list-ctrl))
+  (set! (-> *ATX-settings* list-ctrl func) atx-list-art-group-func)
+  (atx-add-common-group)
+  (atx-add-level-group) 
+  (atx-start)
+)
```
[the remark this answers is not in the document]
Comment (LuminarLight) [reply goal_src/jak2/pc/debug/default-menu-pc.gc]: I removed it and everything seems to be working the same way.

File: goal_src/jak2/pc/debug/anim-tester-x.gc [hunk 2]
```diff
@@ -473,7 +473,12 @@
     (none))
 
   :code (behavior ()
-
+    (local-vars
+        (need-update symbol)
+        (lst atx-list)
+        )
+    (set! need-update #f)
+    (set! lst #f)
```
[the remark this answers is not in the document]
Comment (LuminarLight) [reply goal_src/jak2/pc/debug/anim-tester-x.gc]: Ok, I have done this. But I am curious, what is wrong with local-vars? I can't find documentation about it. Using let is okay for me, I just want to learn.

File: goal_src/jak2/pc/debug/anim-tester-x.gc [hunk 3]
```diff
@@ -532,6 +538,26 @@
...
+            (cpad-clear! 0 right)
+            (+! (-> lst selection) 1)
+            (set! need-update #t))
+          ((< (-> lst  selection) 0)
+            (set! (-> lst selection) (1- (atx-list-size lst))))
+          ((>= (-> lst selection) (atx-list-size lst))
+            (set! (-> lst selection) 0))
+          (need-update
+            ((-> lst func) (atx-list-get-by-index lst (-> lst  selection)))
+            (set! need-update #f))
```
[the remark this answers is not in the document]
Comment (LuminarLight) [reply goal_src/jak2/pc/debug/anim-tester-x.gc]: Thanks, I wasn't fully aware of how (cond worked. Now I understand that it does not evaluate the upcoming conditions if one succeeds. With this new knowledge I also put the "limit checking" into a separate (cond

File: goal_src/jak2/pc/debug/anim-tester-x.gc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ManDude) [reply goal_src/jak2/pc/debug/anim-tester-x.gc]: `local-vars` never initializes a value, it looks ugly, and the scope is difficult to determine.

[the remark this answers is not in the document]
Comment (LuminarLight) [reply goal_src/jak2/pc/debug/anim-tester-x.gc]: Thank you for the explanation.

**Review by ManDude — APPROVED** [review]

**Comment by LuminarLight** [comment]
I see this was approved, can we wait with the merge until I add the same changes to Jak 1? 👀
I should be able to do it tomorrow.

File: goal_src/jak1/pc/debug/anim-tester-x.gc [hunk 4]
```diff
@@ -464,73 +464,100 @@
...
+        (set! *camera-read-buttons* #f)
         )
     (when (!= *master-mode* 'menu)
       (add-debug-x #t (bucket-id debug-no-zbuf) (-> self root trans) *color-white*)
       )
     (none))
 
   :code (behavior ()
+    (let
+      ((need-update (the-as symbol #f))
```
Comment (ManDude) [inline goal_src/jak1/pc/debug/anim-tester-x.gc:476]: `#f` is already a symbol, so you don't need this cast.

**Review by ManDude — APPROVED** [review]

File: goal_src/jak1/pc/debug/anim-tester-x.gc (hunk 4 above)
Comment (LuminarLight) [reply goal_src/jak1/pc/debug/anim-tester-x.gc]: Fair point. Makes sense.
I now fixed this for both games.

**Comment by xTVaser** [comment]
Is this good to merge? there's not really a need to continually merge with master unless there are merge conflicts.

**Comment by xTVaser** [comment]
ok to test

**Pull request merged**